File: C++/26-remove-duplicates-from-sorted-array.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int slowIndex = 0;
        for (int fastIndex = 1; fastIndex < nums.size(); fastIndex++) {
            if (nums[slowIndex] != nums[fastIndex]) {
                nums[++slowIndex] = nums[fastIndex];
            }
        }
        return ++slowIndex;
    }
};

File: JS/26-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let lo = 0;
    for (let hi = 1; hi < nums.length; hi++) {
        if (nums[hi] != nums[lo]) nums[++lo] = nums[hi];
    }
    return lo + 1;
};